### PR TITLE
Move to kotlin dsl and indra

### PR DESCRIPTION
**Pull request by broccolai**
Changes all Gradle to kotlin DSL and uses Indra.
Javadoc usages of > were fixed.
Also switched to the bstats maven dependency so null checking wouldn't cry about the file.

Publishing needs to be double-checked, Indra publishing could be set up instead of the maven publishing plugin but I couldn't get it working.
Also bumped dependencies where safe

### CODE REVIEW

File: buildSrc/src/main/kotlin/Versions.kt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  // CarbonChat
+  const val CARBON_BASE = "1.0"
+
+  // Plugins
+  const val INDRA = "1.3.1"
+  const val CHECKER_PLUGIN = "0.5.16"
+  const val GRGIT = "4.1.0"
+
+  // Global
+  const val CHECKER_QUAL = "3.7.0"
```
Comment (jpenilla) [inline buildSrc/src/main/kotlin/Versions.kt:11]: 3.11.0

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.7.1-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.8.2-bin.zip
```
Comment (jpenilla) [inline gradle/wrapper/gradle-wrapper.properties:3]: 6.8.3

File: CarbonChat-API/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,42 @@
+description = "CarbonChat-API"
```
Comment (jpenilla) [inline CarbonChat-API/build.gradle.kts:1]: Whats the point of setting the description to the same thing as the subproject name?

**Review by jpenilla — CHANGES_REQUESTED** [review]

File: CarbonChat-Bukkit/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    dependsOn(withType<ShadowJar>())
+  }
+
+  // Output a single jar
+  withType<ShadowJar> {
+    minimize()
+    archiveFileName.set(project.description + "-" + Versions.CARBON_BASE + ".jar")
+    destinationDirectory.set(rootProject.getBuildDir())
+    relocate("org.yaml", "net.draycia.carbon.libs.org.yaml")
+    relocate("net.kyori", "net.draycia.carbon.libs.kyori")
```
Comment (jpenilla) [inline CarbonChat-Bukkit/build.gradle.kts:77]: this should be `net.kyori.adventure.text.minimessage` since this branch uses paper's native adventure

**Review by jpenilla — CHANGES_REQUESTED** [review]

File: CarbonChat-API/build.gradle.kts (hunk 3 above)
Comment (broccolai) [reply CarbonChat-API/build.gradle.kts:1]: Not sure, its what the old gradle file did, the publish task uses it as the artifact

Comment (jpenilla) [reply CarbonChat-API/build.gradle.kts:1]: I guess it's fine, but personally I would drop it and just use the project name

File: build.gradle.kts [hunk 5]
```diff
@@ -0,0 +1,54 @@
+import net.kyori.indra.IndraPlugin
+import net.kyori.indra.IndraCheckstylePlugin
+import net.kyori.indra.sonatypeSnapshots
+import org.checkerframework.gradle.plugin.CheckerFrameworkPlugin
+import org.checkerframework.gradle.plugin.CheckerFrameworkExtension
+
+plugins {
+  `maven-publish`
+  id("net.kyori.indra") version Versions.INDRA
```
Comment (Draycia) [inline build.gradle.kts:9]: Unable to compile with java 8, setting my java version to 11 allows the plugin to compile.

**Review by Draycia — CHANGES_REQUESTED** [review]

**Comment by Draycia** [comment]
oops.

**Pull request merged**